plugins/formal-verification/formver.core/src/org/jetbrains/kotlin/formver/embeddings/TypeEmbedding.kt

```diff
@@ -65,7 +75,8 @@ interface TypeEmbedding {
     fun accessInvariants(): List<TypeInvariantEmbedding> = emptyList()
 
     // Note: this function will replace accessInvariants when nested unfold will be implemented
-    fun predicateAccessInvariant(): TypeInvariantEmbedding? = null
+    // Note: for the moment, unique predicates are not used and so TYPE_PREDICATE can be the default value
```

Comment:
Are these sufficiently similar to warrant having the same function?  Is there any case where you may want one or the other?  It feels like all the implementations will start with a branch.

Reply:
Do you mean having two functions `TypePredicateAccessInvariant` and `UniquePredicateAccessInvariant`, right?

I'm still not sure about what's the best thing to do here. Probably it will be clearer after adding more functionalities to unique predicates. I think that we can keep this for the moment and decide whether to split it or not in next PR.